src/unity/python/turicreate/data_structures/sarray.py

```diff
@@ -353,6 +353,12 @@ def __init__(self, data=[], dtype=None, ignore_cast_failure=False, _proxy=None):
             self.__proxy__ = data.__proxy__
         else:
             self.__proxy__ = UnitySArrayProxy()
+
+            # Guard against numpy.matrix, which cannot be traversed recursively,
+            # since subscripting a matrix always yields another (2d) matrix.
+            if HAS_NUMPY and isinstance(data, numpy.matrix):
+                data = numpy.array(data)
```

Comment:
I recommend using `data = numpy.asarray(data)` here. I think this is the recommended way to go from a matrix to an array. This is *much* faster on large data. In my measurements, `numpy.array` scales linearly with data size, while `numpy.asarray` is constant (it should be constant, since it should just re-cast the same memory as an array).